pkg/integrations/configrequest/handler.go

```diff
@@ -0,0 +1,40 @@
+package configrequest
+
+import (
+	"github.com/newrelic/infrastructure-agent/internal/integrations/v4/integration"
+	"github.com/newrelic/infrastructure-agent/pkg/databind/pkg/databind"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/configrequest/protocol"
+	"github.com/newrelic/infrastructure-agent/pkg/log"
+)
+
+var (
+	// helper for testing purposes
+	NoopHandleFn = func(configProtocol protocol.ConfigProtocol) {}
+)
+
+type Entry struct {
+	Definition integration.Definition
+	YAMLConfig databind.YAMLConfig
+}
+
+type HandleFn func(cfgProtocol protocol.ConfigProtocol)
+
+// NewHandleFn creates a handler func that runs every command within the request batch independently.
+// Each command is run in parallel and won't depend on the results of the other ones.
+func NewHandleFn(configProtocolQueue chan<- Entry, il integration.InstancesLookup, logger log.Entry) HandleFn {
+	return func(cfgProtocol protocol.ConfigProtocol) {
+		cfgRequest := cfgProtocol.BuildConfigRequest()
+		for _, ce := range cfgProtocol.Integrations() {
+			def, err := integration.NewDefinition(ce, il, nil, nil)
+			if err != nil {
+				logger.
+					WithField("config_protocol_version", cfgProtocol.Version()).
+					WithField("name", cfgRequest.ConfigName).
+					WithError(err).
+					Warn("cannot create handler for config protocol")
+				return
+			}
+			configProtocolQueue <- Entry{def.WithConfigRequest(cfgRequest), cfgProtocol.GetConfig()}
```

Comment:
by making an entry per integration definition, will we not multiply the number of calls to databind by the number of definitions? Is it something we should be worried about?

Reply:
I'm reading the code again and it looks like databind.Fetch is called on each integration run anyway, so I think we can disregard this comment

Reply:
I haven't check deeply the databind resolution, but we can have a look after